Kinvey/KinveyTests/SyncStoreTests.swift

```diff
@@ -7919,4 +7919,568 @@ class SyncStoreTests: StoreTestCase {
+        let name1 = UUID().uuidString
+        let age1 = Int(arc4random())
+        let creator1 = UUID().uuidString
+        let lmt1 = Date().toString()
+        let ect1 = Date().toString()
+        
+        let id2 = UUID().uuidString
+        let name2 = UUID().uuidString
+        let age2 = Int(arc4random())
+        let creator2 = UUID().uuidString
+        let lmt2 = Date().toString()
+        let ect2 = Date().toString()
+        
+        let mockObjs: [[String : Any]] = [
+            [
+                "_id": id1,
+                "name": name1,
+                "age": age1,
+                "references" : [reference],
+                "_acl": [
+                    "creator": creator1
+                ],
+                "_kmd": [
+                    "lmt": lmt1,
+                    "ect": ect1
+                ]
+            ],
+            [
+                "_id": id2,
+                "name": name2,
+                "age": age2,
+                "references" : [reference],
+                "_acl": [
+                    "creator": creator2
+                ],
+                "_kmd": [
+                    "lmt": lmt2,
+                    "ect": ect2
+                ]
+            ]
+        ]
+        
+        var count = 0
+        mockResponse { request in
+            let urlComponents = URLComponents(url: request.url!, resolvingAgainstBaseURL: false)!
+            switch urlComponents.path {
+            case "/appdata/\(self.client.appKey!)/\(PersonCodable.collectionName())/":
+                return HttpResponse(json: mockObjs)
+            default:
+                XCTFail(request.url!.path)
+                return HttpResponse(statusCode: 404, data: Data())
+            }
+        }
+        defer {
+            setURLProtocol(nil)
+        }
+        
+        let store = try! DataStore<PersonCodable>.collection(.sync)
+        let realm = (store.cache!.cache as! RealmCache<PersonCodable>).realm
+        let items = try! store.pull(options: nil).waitForResult(timeout: defaultTimeout).value()
+        XCTAssertEqual(mockObjs.count, items.count)
+        XCTAssertEqual(mockObjs.count, store.cache!.count(query: nil))
+        XCTAssertEqual(1, realm.objects(Reference.self).count)
+        
+        if let person = items.first {
+            let count = try! store.remove(person).waitForResult(timeout: defaultTimeout).value()
+            XCTAssertEqual(1, count)
+            XCTAssertEqual(1, realm.objects(Reference.self).count)
+        }
+        
+        if let person = items.last {
+            XCTAssertEqual(person.references.first?.entityId, referenceId)
+            let refreshedInstance = try! store.find(person.entityId!).waitForResult(timeout: defaultTimeout).value()
+            XCTAssertEqual(refreshedInstance.references.first?.entityId, referenceId)
+        }
+    }
+    
+    func testCascadeDeleteItemsWithOtherReferencesInAList() {
+        let referenceId = UUID().uuidString
```

Comment:
What is the difference between this function and the previous `func testCascadeDeleteItemsWithOtherReferencesInALists()`?

Reply:
the one above is a scenario where you have 2 lists, this one is a scenario where you have an object and a list